src/graphnet/data/utilities/parquet_to_sqlite.py

```diff
@@ -0,0 +1,227 @@
+            self._excluded_fields = excluded_fields
+        else:
+            self._excluded_fields = []
+        self._mc_truth_table = mc_truth_table
+        self._event_counter = 0
+        self._created_tables = []
+
+    def _find_files(self, parquet_path: str = None):
+        if isinstance(parquet_path, str):
+            if parquet_path.endswith(".parquet"):
+                files = [parquet_path]
+            else:
+                files = glob.glob(f"{parquet_path}/*.parquet")
+        elif isinstance(parquet_path, list):
+            files = []
+            for path in parquet_path:
+                files.extend(glob.glob(f"{path}/*.parquet"))
+        assert len(files) > 0, f"No Files Found in {parquet_path}"
+        return files
+
+    def run(self, outdir: str = None, database_name: str = None):
+        self._setup_directory(outdir, database_name)
+        for i in trange(
+            len(self._parquet_files), desc="Main", colour="#0000ff", position=0
+        ):
+            parquet_file = ak.from_parquet(self._parquet_files[i])
+            n_events_in_file = self._count_events(parquet_file)
+            for j in trange(
+                len(parquet_file.fields),
+                desc="%s" % (self._parquet_files[i].split("/")[-1]),
+                colour="#ffa500",
+                position=1,
+                leave=False,
+            ):
+                if parquet_file.fields[j] not in self._excluded_fields:
+                    self._save_to_sql(
+                        outdir,
+                        database_name,
+                        parquet_file,
+                        parquet_file.fields[j],
+                        n_events_in_file,
+                    )
+            self._event_counter += n_events_in_file
+        self._save_config(outdir, database_name)
+        print(
+            f"Database saved at \n{outdir}/{database_name}/data/{database_name}.db"
+        )
+
+    def _count_events(self, open_parquet_file: str = None):
+        return len(open_parquet_file[self._mc_truth_table])
+
+    def _save_to_sql(
+        self,
+        outdir: str = None,
+        database_name: str = None,
+        ak_array: awkward.Array = None,
+        field_name: str = None,
+        n_events_in_file: int = None,
+    ):
+        df = self._make_df(ak_array, field_name, n_events_in_file)
+        if field_name in self._created_tables:
+            self._submit_to_database(outdir, database_name, field_name, df)
+        else:
+            self._create_table(
+                df, field_name, outdir, database_name, n_events_in_file
+            )
+            self._created_tables.append(field_name)
+            self._submit_to_database(outdir, database_name, field_name, df)
+
+    def _make_df(
+        self,
+        ak_array: awkward.Array = None,
+        field_name: str = None,
+        n_events_in_file: int = None,
+    ):
+        df = pd.DataFrame(ak.to_pandas(ak_array[field_name]))
```

Comment:
Are both `pd.DataFrame(...)` and `ak.to_pandas(...)` necessary?

Reply:
In my experience, yes.